ingestion/src/metadata/ingestion/source/pipeline/airflow/metadata.py

```diff
@@ -249,6 +251,38 @@ def yield_pipeline_status(
                 " Skipping status ingestion."
             )
 
+    def get_schedule_interval(self, pipeline_data) -> Optional[str]:
+        """
+        Fetch Schedule Intervals from Airflow Dags
+        """
+        schedule_interval_timetable_val = pipeline_data.get("timetable", {}).get(
+            "__var", {}
+        )
+        if schedule_interval_timetable_val:
+            # Fetch Cron as String
+            return schedule_interval_timetable_val.get("expression", None)
+        schedule_interval_val = pipeline_data.get("schedule_interval", {})
+        if schedule_interval_val:
+            type_value = schedule_interval_val.get("__type", {})
+            if type_value == TIMEDELTA:
+                var_value = schedule_interval_val.get("__var", {})
+                # types of schedule interval with timedelta
+                # timedelta(days=1) = `1 day, 0:00:00`
+                return str(timedelta(seconds=var_value))
+
+        try:
+            # If the Schedule interval is a const value like @once, @yearly etc
+            # __type sends the module path, and once instantiated
+            return import_from_module(
+                pipeline_data.get("timetable", {}).get("__type", {})
+            )().summary
+        except Exception:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Couldn't fetch schedule interval for dag {pipeline_data.get('_dag_id')}"
+            )
+        return ""
```

Comment:
aiaiai please let's not return empty strings, either something or `None` and we can flag the type hint as `Optional` 🙏